echoshader/dev/echo_gram.py

```diff
@@ -63,42 +185,139 @@ class Echogram(param.Parameterized):
             doc="Select clim",
         )
 
-        color_map = param.String(default="jet", doc="colormap")
+        color_map = param.String(default="jet", doc="Colormap")
 
-        @param.depends("select_date", "select_channel", "range_clim", "color_map")
+        bin_size = param.Integer(24, bounds=(1, 200), doc="Bin size")
+
+        @param.depends(
+            "start_input",
+            "end_input",
+            "select_channel",
+            "range_clim",
+            "color_map",
+            "bin_size",
+        )
         def view(self):
+            start_input_time = self.start_input
+
+            end_input_time = self.end_input
+
+            time_range = slice(start_input_time, end_input_time)
 
-            date_picker = self.select_date
             channel = self.select_channel
+
             clim = self.range_clim
+
             color_map = self.color_map
 
-            start_time = (
-                str(date_picker.year)
-                + "-"
-                + str(date_picker.month)
-                + "-"
-                + str(date_picker.day)
-                + "T00:00:00"
+            bin_size = self.bin_size
+
+            gram = (
+                self.MVBS.Sv.sel(channel=channel, ping_time=time_range)
+                .hvplot(
+                    kind="image",
+                    x="ping_time",
+                    y="echo_range",
+                    title="Sv : " + channel,
+                    cmap=color_map,
+                    clim=clim,
+                    rasterize=True,
+                )
+                .options(invert_yaxis=True)
             )
 
-            end_time = (
-                str(date_picker.year)
-                + "-"
-                + str(date_picker.month)
-                + "-"
-                + str(date_picker.day)
-                + "T23:59:59"
+            bounds, hist, table = echo_stats.plot_hist(
+                gram, self.MVBS.sel(channel=channel, ping_time=time_range), bin_size
             )
 
-            time_range = slice(start_time, end_time)
+            return panel.Column(gram * bounds, hist, table)
+
+    return EchogramHist()
+
+
+def echogram_map(MVBS):
+    """Get an echogram and map with control widgets
+
+    Parameters
+    ----------
+    MVBS : xr.Dataset
+        MVBS_ds data set or ds_Sv data array
+
+    Returns
+    -------
+    type:
+        param class
+    describe:
+        return values contains echogram and control widgets
+        use 'panel.Row(echogram(MVBS).param,echogram(MVBS).view)'
+        to get a complete panel
+        echogram(MVBS).param shows param(control widgets)
+        echogram(MVBS).view shows echogram image
```

Comment:
Same numpy docstring fix here. See https://numpydoc.readthedocs.io/en/latest/format.html#returns

type for this should be param.Parametrize